### PR TITLE
Refactor graphstorm.sagemaker logging printing

### PR DESCRIPTION
**Pull request by classicsong**
We refactor graphstorm.sagemaker module to use python logging package to print information.
The default logging level is INFO.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/sagemaker/sagemaker_infer.py [hunk 1]
```diff
@@ -247,7 +255,8 @@ def run_infer(args, unknownargs):
 
     # Download Saved model
     download_model(model_artifact_s3, model_path, sagemaker_session)
-    print(f"{model_path} {os.listdir(model_path)}")
+    logging.info("Successfully download the model into %s.\n The model has: %s.",
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/sagemaker_infer.py:258]: Do we want to say "The model files are..." here?

File: python/graphstorm/sagemaker/sagemaker_infer.py [hunk 2]
```diff
@@ -295,12 +304,12 @@ def run_infer(args, unknownargs):
...
         # clean embs, so SageMaker does not need to upload embs again
         remove_embs(emb_path)
-        print("Worker End")
+        logging.info("Worker End")
 
     sock.close()
     if err_code != 0:
         # Report an error
-        print("Task failed")
+        logging.info("Task failed")
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/sagemaker_infer.py:312]: Maybe we want to elevate this to an error log level?

File: python/graphstorm/sagemaker/sagemaker_train.py [hunk 3]
```diff
@@ -167,8 +167,8 @@ def run_train(args, unknownargs):
     # start the ssh server
     subprocess.run(["service", "ssh", "start"], check=True)
 
-    print(f"Know args {args}")
-    print(f"Unknow args {unknownargs}")
+    logging.info("Know args %s", args)
+    logging.info("Unknow args %s", unknownargs)
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/sagemaker_train.py:171]:
```suggestion
    logging.info("Known args %s", args)
    logging.info("Unknown args %s", unknownargs)
```

File: python/graphstorm/sagemaker/utils.py [hunk 4]
```diff
@@ -270,32 +271,41 @@ def download_graph(graph_data_s3, graph_name, part_id, local_path, sagemaker_ses
...
+            logging.info("Download graph remap from %s to %s",
+                         file, graph_path)
             S3Downloader.download(file, graph_path,
                                   sagemaker_session=sagemaker_session)
         except Exception: # pylint: disable=broad-except
-            print(f"node id remap file {file} does not exist")
+            logging.warning("node id remap file %s does not exist", file)
 
-    print(f"Finish download graph data from {graph_data_s3}")
+    logging.info("Finish download graph data from %s", graph_data_s3)
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/utils.py:308]:
```suggestion
    logging.info("Finished downloading graph data from %s", graph_data_s3)
```

File: python/graphstorm/sagemaker/utils.py [hunk 5]
```diff
@@ -335,7 +345,7 @@ def upload_model_artifacts(model_s3_path, model_path, sagemaker_session):
     sagemaker_session: sagemaker.session.Session
         sagemaker_session to run download
     """
-    print(f"Upload model artifacts to {model_s3_path}")
+    logging.info("Upload model artifacts to %s", model_s3_path)
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/utils.py:348]:
```suggestion
    logging.info("Uploading model artifacts to %s", model_s3_path)
```

File: python/graphstorm/sagemaker/sagemaker_infer.py [hunk 6]
```diff
@@ -174,8 +176,8 @@ def run_infer(args, unknownargs):
     # start the ssh server
     subprocess.run(["service", "ssh", "start"], check=True)
 
-    print(f"Know args {args}")
-    print(f"Unknow args {unknownargs}")
+    logging.info("Know args %s", args)
+    logging.info("Unknow args %s", unknownargs)
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/sagemaker_infer.py:180]:
```suggestion
    logging.info("Known args %s", args)
    logging.info("Unknown args %s", unknownargs)
```

File: python/graphstorm/sagemaker/sagemaker_train.py [hunk 7]
```diff
@@ -232,8 +238,8 @@ def run_train(args, unknownargs):
     if model_checkpoint_s3 is not None:
         # Download Saved model checkpoint to resume
         download_model(model_checkpoint_s3, restore_model_path, sagemaker_session)
-        print(f"{restore_model_path} {os.listdir(restore_model_path)}")
-
+        logging.info("Successfully download the model into %s.\n The model has: %s.",
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/sagemaker_train.py:241]:
```suggestion
        logging.info("Successfully downloaded the model into %s.\n The model has: %s.",
```

File: python/graphstorm/sagemaker/utils.py [hunk 8]
```diff
@@ -270,32 +271,41 @@ def download_graph(graph_data_s3, graph_name, part_id, local_path, sagemaker_ses
...
+                     graph_part_path)
         S3Downloader.download(os.path.join(graph_data_s3, graph_part),
             graph_part_path, sagemaker_session=sagemaker_session)
-    except Exception: # pylint: disable=broad-except
-        print(f"Can not download graph_data from {graph_data_s3}.")
-        raise RuntimeError(f"Can not download graph_data from {graph_data_s3}.")
+    except Exception as err: # pylint: disable=broad-except
+        logging.error("Can not download graph_data from %s, %s.",
+                      graph_data_s3, str(err))
+        raise RuntimeError(f"Can not download graph_data from {graph_data_s3}, {err}.")
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/utils.py:282]: Do we need both the log and runtime error here? Maybe add all info to the error?

File: python/graphstorm/sagemaker/utils.py [hunk 9]
```diff
@@ -270,32 +271,41 @@ def download_graph(graph_data_s3, graph_name, part_id, local_path, sagemaker_ses
...
     # Try to download node id mapping file if any
     try:
+        logging.info("Download graph id mapping from %s to %s",
+                     os.path.join(graph_data_s3, node_id_mapping),
+                     graph_path)
         S3Downloader.download(os.path.join(graph_data_s3, node_id_mapping),
             graph_path, sagemaker_session=sagemaker_session)
     except Exception: # pylint: disable=broad-except
-        print("node id mapping file does not exist")
+        logging.warning("node id mapping file does not exist")
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/utils.py:294]: Is this expected to happen in some occasions? Can we add some more context (like node type) to the message?

File: python/graphstorm/sagemaker/utils.py [hunk 10]
```diff
@@ -314,9 +324,9 @@ def upload_data_to_s3(s3_path, data_path, sagemaker_session):
     try:
         ret = S3Uploader.upload(data_path, s3_path,
             sagemaker_session=sagemaker_session)
-    except Exception: # pylint: disable=broad-except
-        print(f"Can not upload data into {s3_path}")
-        raise RuntimeError(f"Can not upload data into {s3_path}")
+    except Exception as err: # pylint: disable=broad-except
+        logging.error("Can not upload data into %s", s3_path)
+        raise RuntimeError(f"Can not upload data into {s3_path}. {err}")
```
Comment (thvasilo) [inline python/graphstorm/sagemaker/utils.py:329]: Same as above, we can skip the log and add all info to error.

**Review by thvasilo — APPROVED** [review]
LGMT, minor suggestions.

File: python/graphstorm/sagemaker/utils.py (hunk 8 above)
Comment (classicsong) [reply python/graphstorm/sagemaker/utils.py:282]: I am not sure how sagemaker capture the runtime error.
Let's keep both.

File: python/graphstorm/sagemaker/utils.py (hunk 9 above)
Comment (classicsong) [reply python/graphstorm/sagemaker/utils.py:294]: It is possible that users do not provide the node id mapping file.

Comment (classicsong) [reply python/graphstorm/sagemaker/utils.py:294]: Let me add more info.

**Pull request merged**